src/test/java/commission/services/PersonServiceTest.java

```diff
@@ -0,0 +1,126 @@
+package commission.services;
+
+import commission.dao.PersonDaoImpl;
+import commission.entity.Person;
+import commission.service.PersonServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class PersonServiceTest {
+
+    @Mock
+    private PersonDaoImpl personDao;
+
+    @InjectMocks
+    private PersonServiceImpl personService;
+
+    @BeforeEach
+    public void setUp() {
+        when(personDao.findById(1)).thenReturn(Optional.empty());
+        when(personDao.findById(2)).thenReturn(Optional.of(new Person(2, "Levi", "Ackerman")));
+        when(personDao.findById(3)).thenReturn(Optional.of(new Person(2, "Mikasa", "Ackerman")));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        Mockito.reset(personDao);
+    }
+
+    @Test
+    void whenSearchingForAll_findAll_thenReturnAll(){
```

Comment:
Regarding your question on test naming. I tend to use a `given_when_then` naming convention so something like `givenFindAll_whenDaoReturnsMultipleRecords_thenReturnMultipleRecords`. The way I structure it is `given<methodName>_when<whatHappensInTheTest>_then<whatIsTheExpectedOutcome>` you can lookup "semantic test naming" if you want to read a bit about why this is important.

There are many different naming conventions to use depending on the framework, language and company you work in / at. Find something you like and that gives a good description of what the test is supposed to do and then just be consistent with it.